x/stablestake/keeper/tvl.go

```diff
@@ -23,3 +25,19 @@ func (k Keeper) AllTVL(ctx sdk.Context) osmomath.BigDec {
 	}
 	return tvl
 }
+
+func (k Keeper) GetTotalAndPerDenomTVL(ctx sdk.Context) (totalTVL osmomath.BigDec, denomTVL sdk.Coins, err error) {
+	allPools := k.GetAllPools(ctx)
+	totalTVL = osmomath.ZeroBigDec()
+	denomTVL = sdk.Coins{}
+	for _, pool := range allPools {
+		poolTVL := k.TVL(ctx, pool.Id)
+		totalTVL = totalTVL.Add(poolTVL)
+		assetInfo, found := k.oracleKeeper.GetAssetInfo(ctx, pool.DepositDenom)
+		if !found {
+			return osmomath.BigDec{}, sdk.Coins{}, fmt.Errorf("asset info %s not found", pool.DepositDenom)
+		}
+		denomTVL = denomTVL.Add(sdk.NewCoin(assetInfo.Display, poolTVL.Dec().TruncateInt()))
```

Comment:
the conversion to display should be done in query function not keeper function. Also this will fail for OM token